### ISSUE TITLE
Cannot view clicked topic in topic-article 2.5+

### ISSUE THREAD

**Issue by ogdabou** [DemocracyOS/democracyos#1423]
Hi,

I'm currently trying to develop custom things on democracyOS.   
I tested locally with my custom version (minor changes), and on Heroku using the preconfigured app.

**Heroku** *Admin credentials can be sent*

Reproduce:
- go to url : https://textaxel.herokuapp.com/topic/59355e3189141500043b6f7f
- click on topic "a"
- tested on 2.5.0 and 2.5.2(master)

Observe:
- Url is changed correctly
- topic-article view isn't updated as it should be

**My Local tests**
I run local tests using docker, on linux (ubuntu 16 based distro).
I set `emailValidate` to `true`so I don't care about email validation using the test user
```bash
# Build image
git clone https://github.com/DemocracyOS/democracyos.git
git reset --hard 2.5.0
docker build -t democracyos .
# Run mongodb
docker run -e "MONGODB_USERNAME=myusername" -e "MONGODB_PASSWORD=mypassword" -e "MONGODB_DBNAME=mydatabase"  -p 27017:27017 --name mongod frodenas/mongodb
# Start DOS
docker run --link mongod -e "STAFF=test@gmail.com" -e "MONGO_URL=mongodb://myusername:mypassword@mongod:27017/mydatabase" -e "ORGANIZATION_NAME=...." -p 10111:3000 democracyos

# If mongo is crying, I simply remove its volume
docker rm -v mongod
```

**Comment by moustachu** [comment]
Hello ! 

After a few investigations, it's definitly a "React routing problem" : the `Link` component is not refreshing the `topic-article` component properly. 

@ogdabou  I found a quick & dirty fix with this :  

- edit `lib/site/topic-layout/sidebar/list/list-item/component.js`
- add to the `Link` component the props `onClick={this.forceUpdate}`

A clean fix is definitly out of my league right now, I'm not familiar with the new React stack and I might broke something ... 

@mjlescano : hey 😉 ! We should get on this cause this bug is present on the master ...   
Happy to help if I an expert can guide me on this.

**Comment by ogdabou** [comment]
Hi @moustachu,

Thanks for your reply, my app is now up and running ;).
I made the modifications based on the `2.5.0 tag`.

Maybe I'll find a clean solution while my React skills grows!

**Comment by mjlescano** [comment]
Hmm, I think I created the bug when added the `TopicConnector`, because it refreshes the view when a topic changes, but not when the URL changes, here:

https://github.com/DemocracyOS/democracyos/blame/master/lib/site/connectors/topic.js#L28

The method `this.fetchTopic()` should also be called on [`componentWillReceiveProps()`](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops)

@ogdabou or @moustachu Want to send a PR? :shipit:

**Comment by moustachu** [comment]
@mjlescano I'm working on it 😉 

Just a quick question : 
if I add `fetchTopic()` to `TopicConnector` the page only refresh the topic **after** a second click on the sidebar (with the selection of the first ...).

```
componentWillReceiveProps (props) {
      this.fetchTopic()
}
```

Any idea ? (I need to understand that to continue coding on DOS)
Thanks in advance

**Comment by mjlescano** [comment]
@moustachu looks like it's because the moment `componentWillReceiveProps` it's called, the new props are yet not assigned to `this.props`, and `this.fetchTopic` uses `this.props.params.id` to load the Topic.

To fix this, you could make `this.fetchTopic` to take the `id` as an argument, and use it like this: `this.fetchTopic(props.params.id)` or `this.fetchTopic(this.props.params.id)` depending on the context.

**Comment by franciclo** [comment]
this has been fixed in 2.6.2
